homeassistant/components/ialarmxr/__init__.py

```diff
@@ -0,0 +1,94 @@
+"""iAlarmXR integration."""
+import asyncio
+import logging
+
+from async_timeout import timeout
+from pyialarmxr import IAlarmXR
+
+from homeassistant.components.alarm_control_panel import SCAN_INTERVAL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_USERNAME,
+    STATE_UNKNOWN,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, IALARMXR_TO_HASS
+
+PLATFORMS = [Platform.ALARM_CONTROL_PANEL]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up iAlarmXR config."""
+    host = entry.data[CONF_HOST]
+    port = entry.data[CONF_PORT]
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+
+    ialarmxr = IAlarmXR(username, password, host, port)
+
+    try:
+        async with timeout(10):
+            mac = await hass.async_add_executor_job(ialarmxr.get_mac)
+    except (asyncio.TimeoutError, ConnectionError) as ex:
```

Comment:
Uh ok, now I got it, thanks @bdraco 👍 I reverted with the previous implementation raising ConfigEntryNotReady exception. Unfortunately I cannot distinguish a ConnectionError by an authentication error because of the underlying protocol (communication with Antifurto365's p2p service) doesn't raise an authentication exception but a socket timeout...very sad